Aikido.Zen.Core/Patches/WebRequestPatches.cs

```diff
@@ -1,46 +1,156 @@
+                    var prefix = typeof(WebRequestPatches).GetMethod(prefixMethodName, BindingFlags.Static | BindingFlags.NonPublic);
+                    var postfix = typeof(WebRequestPatches).GetMethod(postfixMethodName, BindingFlags.Static | BindingFlags.NonPublic);
+                    harmony.Patch(method, prefix != null ? new HarmonyMethod(prefix) : null, postfix != null ? new HarmonyMethod(postfix) : null);
                 }
             }
             catch
             {
-                // some methods fail to patch (abstract or not implemented) depending on the framework, so we just ignore them
+                // pass through
             }
-
         }
 
-        internal static bool CaptureRequest(WebRequest __instance)
+        /// <summary>
+        /// Harmony prefix patch executed before the original WebRequest method.
+        /// Captures the start time and request details.
+        /// </summary>
+        /// <param name="__instance">The WebRequest instance.</param>
+        /// <param name="__originalMethod">The original method being patched.</param>
+        internal static bool CaptureRequestStart(WebRequest __instance, MethodBase __originalMethod)
         {
-            var (hostname, port) = UriHelper.ExtractHost(__instance.RequestUri);
-            Agent.Instance.CaptureOutboundRequest(hostname, port);
+            try
+            {
+                var (hostname, port) = UriHelper.ExtractHost(__instance.RequestUri);
+                Agent.Instance.CaptureOutboundRequest(hostname, port); // Keep capturing the attempt immediately
+                var operation = __originalMethod?.DeclaringType?.FullName ?? __instance.GetType().FullName;
+                var stopwatch = Stopwatch.StartNew();
+                var state = new RequestState(stopwatch, operation, hostname, port);
+                // Use Add. If the key somehow already exists (unlikely), the Add method will throw,
+                // and the exception will be caught by the outer try-catch block.
+                requestStates.Add(__instance, state);
+            }
+            catch
+            {
+                // pass through
+            }
             return true;
+
+        }
+
+        /// <summary>
+        /// Harmony postfix patch executed after the original WebRequest method completes.
+        /// Calculates duration and reports the inspection results.
+        /// </summary>
+        /// <param name="__instance">The WebRequest instance.</param>
+        internal static void CaptureRequestEnd(WebRequest __instance)
+        {
+            try
+            {
+                if (requestStates.TryGetValue(__instance, out var state))
+                {
+                    state.Stopwatch.Stop();
+                    // TODO: Determine actual values for these based on response or other factors
```

Comment:
I'd leave those out for now, that way we know in core which values are not implemented yet

Reply:
So rm across the entire PR